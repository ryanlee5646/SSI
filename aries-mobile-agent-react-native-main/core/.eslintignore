node_modules
__tests__
ios
android

# generated by bob
lib/
bifold-app/
generate-app-package-json.js