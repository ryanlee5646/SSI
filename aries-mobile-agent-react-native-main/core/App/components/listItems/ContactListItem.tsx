import type { ConnectionRecord } from '@aries-framework/core'

import { StackNavigationProp } from '@react-navigation/stack'
import React from 'react'
import { View, StyleSheet, TouchableOpacity } from 'react-native'

import { dateFormatOptions } from '../../constants'
import { useTheme } from '../../contexts/theme'
import { Screens, SettingStackParams, Stacks } from '../../types/navigators'
import Text from '../texts/Text'
import Title from '../texts/Title'

interface Props {
  contact: ConnectionRecord
  navigation: StackNavigationProp<SettingStackParams, Screens.Settings>
}

const ContactListItem: React.FC<Props> = ({ contact, navigation }) => {
  const { ColorPallet, borderRadius } = useTheme()
  const styles = StyleSheet.create({
    container: {
      marginTop: 15,
      marginHorizontal: 15,
      padding: 10,
      borderRadius,
      backgroundColor: ColorPallet.brand.secondaryBackground,
    },
    date: {
      textAlign: 'right',
    },
  })
  return (
    <TouchableOpacity
      onPress={() =>
        navigation
          .getParent()
          ?.navigate(Stacks.ContactStack, { screen: Screens.Chat, params: { connectionId: contact.id } })
      }
    >
      <View key={contact.id} style={styles.container}>
        <Title>{contact?.alias || contact?.invitation?.label}</Title>
        <Text>{contact.did}</Text>
        <Text style={styles.date}>{contact.createdAt.toLocaleDateString('en-CA', dateFormatOptions)}</Text>
      </View>
    </TouchableOpacity>
  )
}

export default ContactListItem
