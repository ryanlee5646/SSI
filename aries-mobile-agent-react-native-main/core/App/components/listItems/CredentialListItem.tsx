import type { CredentialRecord } from '@aries-framework/core'

import { useNavigation } from '@react-navigation/core'
import { StackNavigationProp } from '@react-navigation/stack'
import React from 'react'
import { useTranslation } from 'react-i18next'
import { TouchableOpacity } from 'react-native'

import { CredentialStackParams, Screens } from '../../types/navigators'
import { testIdWithKey } from '../../utils/testable'
import CredentialCard from '../misc/CredentialCard'

interface CredentialListItemProps {
  credential: CredentialRecord
  revoked?: boolean
}

const CredentialListItem: React.FC<CredentialListItemProps> = ({ credential, revoked = false }) => {
  const { t } = useTranslation()
  const navigation = useNavigation<StackNavigationProp<CredentialStackParams>>()

  return (
    <TouchableOpacity
      accessible={true}
      accessibilityLabel={t('Credentials.CredentialDetails')}
      testID={testIdWithKey('CredentialDetails')}
      onPress={() => navigation.navigate(Screens.CredentialDetails, { credentialId: credential.id })}
    >
      <CredentialCard credential={credential} revoked={revoked} />
    </TouchableOpacity>
  )
}

export default CredentialListItem
