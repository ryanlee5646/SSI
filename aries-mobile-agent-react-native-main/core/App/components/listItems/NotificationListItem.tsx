import type { CredentialRecord, ProofRecord } from '@aries-framework/core'

import { useNavigation } from '@react-navigation/core'
import { StackNavigationProp } from '@react-navigation/stack'
import React from 'react'
import { useTranslation } from 'react-i18next'
import { StyleSheet, View, Text } from 'react-native'
import Icon from 'react-native-vector-icons/MaterialIcons'

import { useTheme } from '../../contexts/theme'
import { GenericFn } from '../../types/fn'
import { HomeStackParams, Screens, Stacks } from '../../types/navigators'
import { parsedSchema } from '../../utils/helpers'
import { testIdWithKey } from '../../utils/testable'
import Button, { ButtonType } from '../buttons/Button'

const iconSize = 30

export enum NotificationType {
  CredentialOffer = 'Offer',
  ProofRequest = 'Proof',
}

interface NotificationListItemProps {
  notificationType: NotificationType
  notification: CredentialRecord | ProofRecord
}

const NotificationListItem: React.FC<NotificationListItemProps> = ({ notificationType, notification }) => {
  const navigation = useNavigation<StackNavigationProp<HomeStackParams>>()
  const { t } = useTranslation()
  const { ColorPallet, TextTheme } = useTheme()
  const styles = StyleSheet.create({
    container: {
      backgroundColor: ColorPallet.notification.info,
      borderColor: ColorPallet.notification.infoBorder,
      borderRadius: 5,
      borderWidth: 1,
      padding: 10,
    },
    headerContainer: {
      flexDirection: 'row',
      paddingHorizontal: 5,
      paddingTop: 5,
    },
    bodyContainer: {
      flexGrow: 1,
      flexDirection: 'column',
      marginLeft: 10 + iconSize,
      paddingHorizontal: 5,
      paddingBottom: 5,
    },
    headerText: {
      ...TextTheme.normal,
      flexShrink: 1,
      fontWeight: 'bold',
      alignSelf: 'center',
      color: ColorPallet.notification.infoText,
    },
    bodyText: {
      ...TextTheme.normal,
      flexShrink: 1,
      marginVertical: 15,
      paddingBottom: 10,
      color: ColorPallet.notification.infoText,
    },
    icon: {
      marginRight: 10,
      alignSelf: 'center',
    },
  })
  let onPress: GenericFn
  let title = ''
  let body = ''

  switch (notificationType) {
    case NotificationType.CredentialOffer:
      // eslint-disable-next-line no-case-declarations
      const { name, version } = parsedSchema(notification as CredentialRecord)
      onPress = () =>
        navigation.getParent()?.navigate(Stacks.NotificationStack, {
          screen: Screens.CredentialOffer,
          params: { credentialId: notification.id },
        })
      title = t('CredentialOffer.CredentialOffer')
      body = `${name} v${version}`
      break
    case NotificationType.ProofRequest:
      title = t('ProofRequest.ProofRequest')
      body = (notification as ProofRecord).requestMessage?.indyProofRequest?.name || ''
      onPress = () =>
        navigation
          .getParent()
          ?.navigate(Stacks.NotificationStack, { screen: Screens.ProofRequest, params: { proofId: notification.id } })
      break
    default:
      throw new Error('NotificationType was not set correctly.')
  }

  return (
    <View style={styles.container}>
      <View style={styles.headerContainer}>
        <View style={[styles.icon]}>
          <Icon name={'info'} size={iconSize} color={ColorPallet.notification.infoIcon} />
        </View>
        <Text style={styles.headerText} testID={testIdWithKey('HeaderText')}>
          {title}
        </Text>
      </View>
      <View style={styles.bodyContainer}>
        <Text style={styles.bodyText} testID={testIdWithKey('BodyText')}>
          {body}
        </Text>
        <Button
          title={t('Global.View')}
          accessibilityLabel={t('Global.View')}
          testID={testIdWithKey('View')}
          buttonType={ButtonType.Primary}
          onPress={onPress}
        />
      </View>
    </View>
  )
}

export default NotificationListItem
