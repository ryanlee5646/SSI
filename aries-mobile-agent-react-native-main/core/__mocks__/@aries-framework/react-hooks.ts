import { CredentialRecord, ProofRecord } from '@aries-framework/core'

const useCredentials = jest.fn().mockReturnValue({ credentials: [] } as any)
const useCredentialByState = jest.fn().mockReturnValue([] as CredentialRecord[])
const useProofByState = jest.fn().mockReturnValue([] as ProofRecord[])
const mockCredentialModule = {
  acceptOffer: jest.fn(),
  declineOffer: jest.fn(),
}
const mockProofModule = {
  getRequestedCredentialsForProofRequest: jest.fn(),
  acceptRequest: jest.fn(),
  declineRequest: jest.fn(),
}
const useAgent = () => ({
  agent: {
    credentials: mockCredentialModule,
    proofs: mockProofModule,
  },
})
const useCredentialById = jest.fn()
const useProofById = jest.fn()
const useConnectionById = jest.fn()

export {
  useAgent,
  useConnectionById,
  useCredentials,
  useCredentialById,
  useCredentialByState,
  useProofById,
  useProofByState,
}
