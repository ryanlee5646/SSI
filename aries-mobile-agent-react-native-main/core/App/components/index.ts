// DEPRECATED: This creates dependency cycles

import Button from './buttons/Button'
import CheckBoxRow from './inputs/CheckBoxRow'
import TextInput from './inputs/TextInput'
import ContactListItem from './listItems/ContactListItem'
import CredentialListItem from './listItems/CredentialListItem'
import NotificationListItem from './listItems/NotificationListItem'
import QRScanner from './misc/QRScanner'
import HighlightTextBox from './texts/HighlightTextBox'
import InfoTextBox from './texts/InfoTextBox'
import Label from './texts/Label'
import Text from './texts/Text'
import Title from './texts/Title'
import ModularView from './views/ModularView'
import SafeAreaScrollView from './views/SafeAreaScrollView'

export {
  CheckBoxRow,
  ContactListItem,
  CredentialListItem,
  Label,
  ModularView,
  Button,
  NotificationListItem,
  QRScanner,
  SafeAreaScrollView,
  TextInput,
  Text,
  Title,
  InfoTextBox,
  HighlightTextBox,
}
