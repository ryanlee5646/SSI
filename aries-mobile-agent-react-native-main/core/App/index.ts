/* eslint-disable import/no-cycle */
import type { OnboardingStyleSheet } from './screens/Onboarding'

import { Agent } from '@aries-framework/core'
import AgentProvider from '@aries-framework/react-hooks'

import indyLedgers from '../configs/ledgers/indy'

import Button, { ButtonType } from './components/buttons/Button'
import CheckBoxRow from './components/inputs/CheckBoxRow'
import ErrorModal from './components/modals/ErrorModal'
import InfoTextBox from './components/texts/InfoTextBox'
import toastConfig from './components/toast/ToastConfig'
import RootStack from './navigators/RootStack'
import OnboardingPages from './screens/OnboardingPages'
import Splash from './screens/Splash'
import Terms from './screens/Terms'

export { LocalStorageKeys } from './constants'
export { initLanguages, initStoredLanguage, translationResources } from './localization'
export { ConfigurationProvider, useConfiguration } from './contexts/configuration'
export { StoreProvider, StoreContext } from './contexts/store'
export { DispatchAction } from './contexts/reducers/store'
export { ThemeProvider, useTheme } from './contexts/theme'
export { ColorPallet } from './theme'
export { theme } from './theme'
export { testIdWithKey } from './utils/testable'
export { Screens } from './types/navigators'
export { createStyles } from './screens/OnboardingPages'

export type { StoreProviderProps } from './contexts/store'
export type { Theme } from './theme'
export type { ConfigurationContext } from './contexts/configuration'
export type { Onboarding as OnboardingState } from './types/state'
export type { GenericFn } from './types/fn'
export type { AuthenticateStackParams } from './types/navigators'
export type { OnboardingStyleSheet }

export {
  indyLedgers,
  Agent,
  AgentProvider,
  Button,
  ButtonType,
  CheckBoxRow,
  InfoTextBox,
  ErrorModal,
  toastConfig,
  RootStack,
  OnboardingPages,
  Splash,
  Terms,
}
