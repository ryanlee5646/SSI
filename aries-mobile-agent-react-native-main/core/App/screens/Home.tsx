import { CredentialRecord, CredentialState } from '@aries-framework/core'
import { useCredentialByState } from '@aries-framework/react-hooks'
import AsyncStorage from '@react-native-async-storage/async-storage'
import { StackScreenProps } from '@react-navigation/stack'
import React, { useMemo } from 'react'
import { useTranslation } from 'react-i18next'
import { FlatList, StyleSheet, View, Text, Dimensions, TouchableOpacity } from 'react-native'

import { NotificationListItem } from '../components'
import { NotificationType } from '../components/listItems/NotificationListItem'
import NoNewUpdates from '../components/misc/NoNewUpdates'
import { LocalStorageKeys } from '../constants'
import { DispatchAction } from '../contexts/reducers/store'
import { useStore } from '../contexts/store'
import { useTheme } from '../contexts/theme'
import { useNotifications } from '../hooks/notifications'
import { HomeStackParams, Screens } from '../types/navigators'
import { Credential as StoreCredentialState } from '../types/state'

const { width } = Dimensions.get('window')
const offset = 25
const offsetPadding = 5

type HomeProps = StackScreenProps<HomeStackParams, Screens.Home>

const Home: React.FC<HomeProps> = ({ navigation }) => {
  const credentials = [
    ...useCredentialByState(CredentialState.CredentialReceived),
    ...useCredentialByState(CredentialState.Done),
  ]
  const { notifications } = useNotifications()
  const { t } = useTranslation()
  const { ColorPallet, TextTheme } = useTheme()
  const [, dispatch] = useStore()
  const styles = StyleSheet.create({
    container: {
      paddingHorizontal: offset,
    },
    rowContainer: {
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'center',
      paddingHorizontal: offset,
    },
    messageContainer: {
      alignItems: 'center',
      justifyContent: 'center',
      marginTop: 35,
      marginHorizontal: 60,
    },
    header: {
      marginTop: offset,
      marginBottom: 20,
    },
    linkContainer: {
      minHeight: TextTheme.normal.fontSize,
      marginTop: 10,
    },
    link: {
      ...TextTheme.normal,
      color: ColorPallet.brand.link,
    },
  })

  useMemo(() => {
    async function init() {
      try {
        const data = await AsyncStorage.getItem(LocalStorageKeys.RevokedCredentials)
        if (data) {
          const revoked: Set<CredentialRecord['id'] | CredentialRecord['credentialId']> = new Set(
            JSON.parse(data) || []
          )
          const credentialState: StoreCredentialState = {
            revoked,
          }
          dispatch({ type: DispatchAction.CREDENTIALS_UPDATED, payload: [credentialState] })
        }
      } catch (error) {
        // TODO:(am add error handling here)
      }
    }
    init()
  }, [])

  const displayMessage = (credentialCount: number) => {
    if (typeof credentialCount === 'undefined' && credentialCount >= 0) {
      throw new Error('Credential count cannot be undefined')
    }

    let credentialMsg

    if (credentialCount === 1) {
      credentialMsg = (
        <Text>
          {t('Home.YouHave')} <Text style={{ fontWeight: 'bold' }}>{credentialCount}</Text> {t('Home.Credential')}{' '}
          {t('Home.InYourWallet')}
        </Text>
      )
    } else if (credentialCount > 1) {
      credentialMsg = (
        <Text>
          {t('Home.YouHave')} <Text style={{ fontWeight: 'bold' }}>{credentialCount}</Text> {t('Home.Credentials')}{' '}
          {t('Home.InYourWallet')}
        </Text>
      )
    } else {
      credentialMsg = t('Home.NoCredentials')
    }

    return (
      <View style={[styles.messageContainer]}>
        {credentialCount === 0 ? (
          <Text style={[TextTheme.headingOne, { marginTop: offset, marginBottom: 35 }]}>{t('Home.Welcome')}</Text>
        ) : null}
        <Text style={[TextTheme.normal, { marginTop: offset, textAlign: 'center' }]}>{credentialMsg}</Text>
      </View>
    )
  }

  return (
    <View>
      <View style={styles.rowContainer}>
        <Text style={[TextTheme.headingThree, styles.header]}>
          {t('Home.Notifications')}
          {notifications?.length ? ` (${notifications.length})` : ''}
        </Text>
        {notifications?.length > 1 ? (
          <TouchableOpacity
            style={styles.linkContainer}
            activeOpacity={1}
            onPress={() => navigation.navigate(Screens.Notifications)}
          >
            <Text style={styles.link}>{t('Home.SeeAll')}</Text>
          </TouchableOpacity>
        ) : null}
      </View>
      <FlatList
        horizontal
        showsHorizontalScrollIndicator={false}
        scrollEnabled={notifications?.length > 0 ? true : false}
        snapToOffsets={[
          0,
          ...Array(notifications?.length)
            .fill(0)
            .map((n: number, i: number) => i * (width - 2 * (offset - offsetPadding)))
            .slice(1),
        ]}
        decelerationRate="fast"
        ListEmptyComponent={() => (
          <View style={{ marginHorizontal: offset, width: width - 2 * offset }}>
            <NoNewUpdates />
          </View>
        )}
        data={notifications}
        keyExtractor={(item) => item.id}
        renderItem={({ item, index }) => (
          <View
            style={{
              width: width - 2 * offset,
              marginLeft: !index ? offset : offsetPadding,
              marginRight: index === notifications?.length - 1 ? offset : offsetPadding,
            }}
          >
            {item.type === 'CredentialRecord' ? (
              <NotificationListItem notificationType={NotificationType.CredentialOffer} notification={item} />
            ) : (
              <NotificationListItem notificationType={NotificationType.ProofRequest} notification={item} />
            )}
          </View>
        )}
      />
      <View style={styles.container}>{displayMessage(credentials.length)}</View>
    </View>
  )
}

export default Home
