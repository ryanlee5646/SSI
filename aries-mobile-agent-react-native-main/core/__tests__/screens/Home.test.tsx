import { CredentialRecord, CredentialState, ProofRecord, ProofState } from '@aries-framework/core'
import { useCredentialByState, useProofByState } from '@aries-framework/react-hooks'
import { useNavigation } from '@react-navigation/core'
import { fireEvent, render } from '@testing-library/react-native'
import React from 'react'
import { FlatList } from 'react-native'
import { create } from 'react-test-renderer'

import { Button, NotificationListItem } from '../../App/components'
import { NotificationType } from '../../App/components/listItems/NotificationListItem'
import Home from '../../App/screens/Home'

describe('displays a home screen', () => {
  it('renders correctly', () => {
    const tree = create(<Home navigation={useNavigation()} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  /**
   * Scenario: Home Screen without any pending notification
   * Given wallet has successfully loaded
   * When the holder selects the "Home" button in the main navigation bar
   * Then the Home Screen is displayed
   * TODO:(jl) Good enough to be captured by the snapshot?
   */
  it('defaults to no notifications', async () => {
    const { findByText } = render(<Home navigation={useNavigation()} />)
    const notificationLabel = await findByText('Home.NoNewUpdates')

    expect(notificationLabel).toBeTruthy()
  })
})

describe('with a notifications module, when an issuer sends a credential offer', () => {
  const testCredentialRecords: CredentialRecord[] = [
    new CredentialRecord({
      threadId: '1',
      state: CredentialState.OfferReceived,
    }),
  ]
  const testProofRecords: ProofRecord[] = [
    new ProofRecord({
      threadId: '2',
      state: ProofState.RequestReceived,
    }),
  ]

  beforeEach(() => {
    jest.clearAllMocks()
    // @ts-ignore
    useCredentialByState.mockReturnValue(testCredentialRecords)
    // @ts-ignore
    useProofByState.mockReturnValue(testProofRecords)
  })

  /**
   * Scenario: Home Screen with pending notifications
   * Given Wallet has successfully loaded
   * When the Home Screen successfully loads
   * Then the number of pending notifications is displayed in the "Home" button in the main navigation bar
   */
  it('notification label is displayed with number of notifications', async () => {
    const { findByText } = render(<Home navigation={useNavigation()} />)
    const notificationLabel = await findByText('Home.Notifications (2)')

    expect(notificationLabel).toBeTruthy()
  })

  it('Pressing the "See All" button navigates correctly', async () => {
    const navigation = useNavigation()
    const { findByText } = render(<Home navigation={useNavigation()} />)
    const seeAllButton = await findByText('Home.SeeAll')

    expect(seeAllButton).toBeTruthy()

    fireEvent(seeAllButton, 'press')

    expect(navigation.navigate).toBeCalledWith('Notifications')
  })

  /**
   * Scenario: Holder receives a credential offer
   * Given the holder has a connection with an issuer
   * When the issuer sends a credential offer
   * Then the credential offer will arrive in the form of a notification in the home screen
   */
  it('notifications are displayed', () => {
    const tree = create(<Home navigation={useNavigation()} />)
    const root = tree.root
    const flatListInstance = root.findByType(FlatList)

    expect(flatListInstance.findAllByType(NotificationListItem)).toHaveLength(2)
  })

  /**
   * Scenario: Holder selects a credential offer
   * Given the holder has received a credential offer from a connected issuer
   * When the holder selects the credential offer
   * When the holder is taken to the credential offer screen/flow
   */
  it('touch notification triggers navigation correctly', async () => {
    const tree = create(<Home navigation={useNavigation()} />)
    const root = tree.root
    const notifications = root.findAllByType(NotificationListItem)
    const credentialOffer = notifications.find(
      (notification) => !!(notification.props.notificationType === NotificationType.CredentialOffer)
    )
    const button = credentialOffer?.findByType(Button)
    const navigation = useNavigation()

    expect(button).toBeDefined()

    button?.props.onPress()

    expect(navigation.navigate).toHaveBeenCalledTimes(1)
    expect(navigation.navigate).toHaveBeenCalledWith('Notifications Stack', {
      screen: 'Credential Offer',
      params: { credentialId: testCredentialRecords[0].id },
    })
  })

  /**
   * Scenario: Holder selects a proof request
   * Given the holder has received a proof request from a connected verifier
   * When the holder selects the proof request
   * When the holder is taken to the proof request screen/flow
   */
  it('touch notification triggers navigation correctly', async () => {
    const tree = create(<Home navigation={useNavigation()} />)
    const root = tree.root
    const notifications = root.findAllByType(NotificationListItem)
    const proofRequest = notifications.find(
      (notification) => !!(notification.props.notificationType === NotificationType.ProofRequest)
    )
    const button = proofRequest?.findByType(Button)
    const navigation = useNavigation()

    expect(button).toBeDefined()

    button?.props.onPress()

    expect(navigation.navigate).toHaveBeenCalledTimes(1)
    expect(navigation.navigate).toHaveBeenCalledWith('Notifications Stack', {
      screen: 'Proof Request',
      params: { proofId: testProofRecords[0].id },
    })
  })
})
