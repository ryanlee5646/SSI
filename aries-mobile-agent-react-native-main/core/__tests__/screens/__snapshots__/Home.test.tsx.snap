// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays a home screen renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 25,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 26,
            "fontWeight": "bold",
          },
          Object {
            "marginBottom": 20,
            "marginTop": 25,
          },
        ]
      }
    >
      Home.Notifications
      
    </Text>
  </View>
  <RCTScrollView
    ListEmptyComponent={[Function]}
    data={Array []}
    decelerationRate="fast"
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEnabled={false}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    snapToOffsets={
      Array [
        0,
      ]
    }
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        style={
          Object {
            "marginHorizontal": 25,
            "width": 700,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#313132",
              "borderColor": "#0099FF",
              "borderRadius": 5,
              "borderWidth": 1,
              "padding": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "marginRight": 10,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#0099FF",
                      "fontSize": 30,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                î¢Ž
              </Text>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "normal",
                  },
                ]
              }
              testID="com.ariesbifold:id/NoNewUpdates"
            >
              Home.NoNewUpdates
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "paddingHorizontal": 25,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginHorizontal": 60,
            "marginTop": 35,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 38,
              "fontWeight": "bold",
            },
            Object {
              "marginBottom": 35,
              "marginTop": 25,
            },
          ]
        }
      >
        Home.Welcome
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "normal",
            },
            Object {
              "marginTop": 25,
              "textAlign": "center",
            },
          ]
        }
      >
        Home.NoCredentials
      </Text>
    </View>
  </View>
</View>
`;
